### ISSUE TITLE
Set SingleDomainLoaderOptimization to false for AutoMLService

### ISSUE THREAD

**Issue by LittleLittleCloud** [dotnet/machinelearning-modelbuilder#834]
https://devdiv.visualstudio.com/DevDiv/_git/VSUnitTesting?path=%2Fsrc%2FTestWindow%2FSetup%2FTestDiscoveryListener.servicehub.service.json&version=GBmaster&_a=contents

**Comment by jessewinkler** [comment]
@LittleLittleCloud I cannot access the link of how to close the option, I get 401 unauthorized

**Comment by LittleLittleCloud** [comment]
> @LittleLittleCloud I cannot access the link of how to close the option, I get 401 unauthorized

The fix is on ModelBuilder side, I'm working on a fix, and once it's done, I will put a link here and you can download and install.

**Comment by jessewinkler** [comment]
thanks for the explanation, looking forward to trying it out!

**Comment by LittleLittleCloud** [comment]
https://test1storageafyok6j49.blob.core.windows.net/grantsetfix/Microsoft.ML.ModelBuilder.Gallery.vsix

**Comment by jessewinkler** [comment]
The fix allowed training to proceed for me. Thank you!

However, it does seem like training is taking significantly longer that it was before. Maybe this is unrelated, and I haven't done a lot of runs yet but though it worth mentioning.

**Comment by LittleLittleCloud** [comment]
> The fix allowed training to proceed for me. Thank you!
> 
> However, it does seem like training is taking significantly longer that it was before. Maybe this is unrelated, and I haven't done a lot of runs yet but though it worth mentioning.

Do you mean the gap between you click on training and the result of first iteration comes out?

**Comment by jessewinkler** [comment]
Yes, there is definitely a increase in time in that stage. The Model Builder recommends 10 seconds for the training period (for smaller data sets) but the first model build doesn't get out for anywhere from 15 to 140+ seconds. As an example, I took a 2 MB data set and it recommended 10 seconds, in which time it could not train any mode. I increased the limit to 100 seconds and it took 13.3 seconds for AveragedPerceptronOva and 21.6 seconds for SdcaMaximumEntropyMulti and stopped. I don't have much historical data on build times before the hotfix was applied, but it seemed odd that no models can be built in the "recommended" time to train now.

**Comment by LittleLittleCloud** [comment]
Hmmmm, Thanks for the update. We were aware that one of the drawbacks for disabling multi domain optimization is perf, but I don't know the perf loss will be that huge. We will keep looking into to see if there's another solution

**Comment by W1zb1t** [comment]
I seem to be getting a similar issue. In the autoML Log:
`06/28/2020 19:49:36 GMT Standard Time : Error : Error creating Microsoft.ML.ModelBuilder.AutoMLService.AutoMLService instance of CLR service module 'MicrosoftMLModelBuilderAutoML': System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. HResult='-2146232828' 
...
===InnerException===
System.IO.FileLoadException: Loading this assembly would produce a different grant set from other instances. (Exception from HRESULT: 0x80131401) HResult='-2146233343' `

Updating with the VIX above gives me a different error. Runtime exception:
`2020-06-28 20:12:25.0912 DEBUG Training failed with the exception: 
  System.InvalidOperationException: Splitter/consolidator worker encountered exception while consuming source data ---> 
    System.InvalidOperationException: Splitter/consolidator worker encountered exception while consuming source data ---> 
      System.FormatException: Parsing failed with an exception: Could not parse value unknown in line 24875, column Plugin ---> 
        System.InvalidOperationException: Could not parse value unknown in line 24875, column Plugin
   at Microsoft.ML.Data.TextLoader.Parser.ProcessOne(FieldSet vs, ColInfo info, ColumnPipe v, Int32 irow, Int64 line)
...`

**Comment by LittleLittleCloud** [comment]
> I seem to be getting a similar issue. In the autoML Log:
> `06/28/2020 19:49:36 GMT Standard Time : Error : Error creating Microsoft.ML.ModelBuilder.AutoMLService.AutoMLService instance of CLR service module 'MicrosoftMLModelBuilderAutoML': System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. HResult='-2146232828' ... ===InnerException=== System.IO.FileLoadException: Loading this assembly would produce a different grant set from other instances. (Exception from HRESULT: 0x80131401) HResult='-2146233343' `
> 
> Updating with the VIX above gives me a different error. Runtime exception:
> `2020-06-28 20:12:25.0912 DEBUG Training failed with the exception: System.InvalidOperationException: Splitter/consolidator worker encountered exception while consuming source data ---> System.InvalidOperationException: Splitter/consolidator worker encountered exception while consuming source data ---> System.FormatException: Parsing failed with an exception: Could not parse value unknown in line 24875, column Plugin ---> System.InvalidOperationException: Could not parse value unknown in line 24875, column Plugin at Microsoft.ML.Data.TextLoader.Parser.ProcessOne(FieldSet vs, ColInfo info, ColumnPipe v, Int32 irow, Int64 line) ...`

Hi @W1zb1t 
it seems to be a different dataset issue, can you create a new issue and share us with the dataset you have. Thanks

**Comment by rymexs** [comment]
I tried this fix and was able to continue training

https://test1storageafyok6j49.blob.core.windows.net/grantsetfix/Microsoft.ML.ModelBuilder.Gallery.vsix

**Comment by lennieharvey** [comment]
happy to test the fix

**Comment by lennieharvey** [comment]
Fix here worked for me

**Comment by bobend212** [comment]
Can someone update the link with the fix please?
